common/src/msgs/data_requests/types.rs

```diff
@@ -237,15 +237,27 @@ impl TryHashSelf for PostDataRequestArgs {
 pub struct DrConfig {
     /// Number of blocks after which a data request is timed out while waiting
     /// for commits.
-    pub commit_timeout_in_blocks:      u64,
+    pub commit_timeout_in_blocks:        NonZero<u8>,
     /// Number of blocks after which a data request is timed out while waiting
     /// for reveals.
-    pub reveal_timeout_in_blocks:      u64,
+    pub reveal_timeout_in_blocks:        NonZero<u8>,
     /// This is the delay before the backup executors are allowed to start
     /// executing the data request.
-    pub backup_delay_in_blocks:        NonZero<u64>,
+    pub backup_delay_in_blocks:          NonZero<u8>,
     /// The maximum size of all the reveals in a data request.
-    pub dr_reveal_size_limit_in_bytes: usize,
+    pub dr_reveal_size_limit_in_bytes:   NonZero<u16>,
+    /// The maximum size of the input for the execution program.
+    pub exec_input_limit_in_bytes:       NonZero<u16>,
+    /// The maximum size of the input for the tally program.
+    pub tally_input_limit_in_bytes:      NonZero<u16>,
+    /// The maximum size of the consensus filter.
+    pub consensus_filter_limit_in_bytes: NonZero<u16>,
+    /// The maximum size of the memo.
+    pub memo_limit_in_bytes:             NonZero<u16>,
+    /// The maximum size of the payback address.
+    pub payback_address_limit_in_bytes:  NonZero<u16>,
```

Comment:
Nit: could be limited  to a u8 probably? I don't think an address would every go beyond 255 bytes.